src/net_processing.cpp

```diff
@@ -1043,6 +1063,25 @@ void PeerManagerImpl::UpdateChainTipSet(std::set<const CBlockIndex*>& chain_tips
     chain_tips.insert(potential_new_tip);
 }
 
+bool PeerManagerImpl::IsBlockInMainOrBestChain(const CBlockIndex& index)
+{
+    AssertLockHeld(cs_main);
+
+    if (m_chainman.ActiveChain().Contains(&index)) return true;
+    if (m_chainman.m_best_header && m_chainman.m_best_header->GetAncestor(index.nHeight) == &index) return true;
+    return false;
+}
+
+bool PeerManagerImpl::IsBlockInChainTipSet(const CBlockIndex& index, const std::set<const CBlockIndex*>& chain_tips)
```

Comment:
Similarly to the previous commit, why don't obtain `GetChainTipSetForPeerNetwork` internally?